file-system/fake/error.go

```diff
@@ -10,9 +12,44 @@ var (
 
 var noSuchFileOrDirectoryError = errgo.New("no such file or directory")
 
-// IsNoSuchFileOrDirectoryError checks for the given error to be
-// noSuchFileOrDirectoryError. This error is returned in case there cannot
-// any file be found as requested.
-func IsNoSuchFileOrDirectoryError(err error) bool {
+// IsNoSuchFileOrDirectory checks for the given error to be
+// noSuchFileOrDirectoryError. This error is returned in case there cannot any
+// file be found as requested.
+func IsNoSuchFileOrDirectory(err error) bool {
+	cause := errgo.Cause(err)
+
+	if cause == nil {
```

Comment:
are you sure this is needed? I would expect the remaining to work even if `cause` is nil.